libs/common/src/common/identity.py

```diff
@@ -10,6 +10,12 @@
 class AbstractUserIdentityProvider(abc.ABC):
     async def get_user_identity(self) -> str: ...
 
+    async def is_internal(self) -> bool:
+        try:
+            return self.get_user_identity().get("user", {}).get("is_internal", False)
```

Comment:
`get_user_identity()` returns a string - plus it needs to  be awaited.

Reply:
We have  a function to decode the identity header:
https://github.com/RedHatInsights/astro-virtual-assistant-v2/blob/main/libs/common/src/common/auth.py#L55

Reply:
Oh thank you! I updated this function to use the decode function. im thinking we should have this check streamlined still. tested locally and it works now